core/rlp_test.go

```diff
@@ -195,3 +200,36 @@ func BenchmarkHashing(b *testing.B) {
 		b.Fatalf("hash wrong, got %x exp %x", got, exp)
 	}
 }
+
+func TestBlockRlpEncodeDecode(t *testing.T) {
+	zeroTime := uint64(0)
+
+	// create a config where Isthmus upgrade is active
+	config := *params.OptimismTestConfig
+	config.ShanghaiTime = &zeroTime
+	config.IsthmusTime = &zeroTime
+
+	block := getBlock(&config, 10, 2, 50)
+
+	blockRlp, err := rlp.EncodeToBytes(block)
+	assert.Nil(t, err)
+
+	var decoded types.Block
+	err = rlp.DecodeBytes(blockRlp, &decoded)
+	assert.Nil(t, err)
+
+	check := func(f string, got, want interface{}) {
+		if equal := reflect.DeepEqual(got, want); equal != true {
+			t.Errorf("%s mismatch", f)
+			t.Errorf("Got: %+v", got)
+			t.Errorf("Want: %+v", want)
+		}
+	}
+
+	// There's an odd inconsistency in the way `ExtraData` field, when it is empty is returned after
```

Comment:
Is this a dead comment?

Reply:
I thought it's a valuable pointer for anyone touching this test in the future, because if there's an attempt made to compare individual fields like I did, they'll step into the same issue. Happy to remove it though.